projects/output-and-error-handling/client/main_test.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+)
+
+// This is the beginning of the test implementation
+// It is not comprehensive by any means
+
+func TestParseResponseBody(t *testing.T) {
+	t.Run("response with valid body", func(t *testing.T) {
+		body := "Today it will be sunny!"
+
+		// io.NopCloser is a non-closing reader
+		response := &http.Response{Body: io.NopCloser(bytes.NewBufferString(body))}
+
+		result, err := parseResponseBody(response)
+
+		if err != nil {
+			t.Errorf("could not parse body")
+		}
+
+		if result != body {
+			t.Errorf("got %q | want %q", body, result)
+		}
```

Comment:
nit: a more common approach to naming variables intests is `expected, got` or `want, got` as in your error message

Reply:
FWIW the _exact_ naming here tends to be a super language-specific style. I would generally expect to see "expected"/"actual" in Java, and "want"/"got" in Go, for instance :)